core/state/state_object.go

```diff
@@ -252,17 +260,29 @@ func (s *stateObject) setState(key, value common.Hash) {
 	s.dirtyStorage[key] = value
 }
 
+// finalise moves all dirty storage slots into the pending area to be hashed or
+// committed later.
```

Comment:
perhaps add a comment about when this is invoked. It's once every transaction, right?

Reply:
Done.